libopenage/unit/action.cpp

```diff
@@ -771,6 +771,15 @@ GatherAction::GatherAction(Unit *e, UnitReference tar)
 	complete{false},
 	target_resource{true},
 	target{tar} {
+
+		Unit *target = this->target.get();
+		if (target->has_attribute(attr_type::resource)) {
+			auto &resource_attr = target->get_attribute<attr_type::resource>();
+			this->resource_type = resource_attr.resource_type;
+
+			//TODO toggle docks
+			this->docks = false;
+		}
```

Comment:
These lines are one indent level too much, they need to be at the same level as `target{tar}`.
What is `resource_type` initialized to when the target doesn't have the resource attribute? You probably wanna add it to the initializer list above or else but i think you need to handle it.